### DISCUSSION TITLE
.env file support

### DISCUSSION THREAD

**Discussion by aaronpowell** [keystonejs/keystone#7176] [Feature requests]
It's common, especially for local development, to use a `.env` file (or environment-specific one) to set environment variables that you're going to need, such as db connection string, API keys, etc.

Keystone currently doesn't have any built-in way in which you can use this file, the options are to use a tool like `env-cmd` to load the file before running `keystone dev`, manually doing `export FOO=bar` as many times as required, using the `dotenv` npm package yourself to load them, or just embedding the values in the .ts files (jokes, I'm totally not doing that 🤫).

I'd like for Keystone to do this for me. If I run `keystone dev`, then I'd expect it to load my `.env.development` overrides for `.env`, if it's `keystone start` then treat it as production (which it already does).

This is how I've found other similar tools operate, so I have been trained to expect it.

**Comment by bkbooth** [comment]
I brought the `dotenv` package in myself and have `import 'dotenv/config'` at the top of my _keystone.ts_ config file which has worked fine since the day I setup my project, so I kinda forgot that Keystone didn't support it.

That said, I came to Keystone expecting it and was slightly surprised it didn't work out of the box. I'd definitely be in favour of adding it by default.

**Reply by aaronpowell** [reply]
Yeah, it's not hard to "work around", but it's the sort of thing that I'd like to see not having to do something specific for.

**Reply by bkbooth** [reply]
Yeah, it makes sense to me. Is there a good reason to _not_ include it? Most of the reasons seem to revolve around it being fairly easy to add it if you want it. But a lot of similar tools _do_ include it by default, almost a de facto standard.

**Reply by nillswe** [reply]
That worked for me. thanks!

**Reply by dcousens** [reply]
@aaronpowell out of interest, what other tooling are you using that includes `dotenv` by default?

**Reply by bkbooth** [reply]
The big 2 that come to mind for me are [Create React App](https://create-react-app.dev/docs/adding-custom-environment-variables/#adding-development-environment-variables-in-env) and [Next.js](https://nextjs.org/docs/basic-features/environment-variables)

**Reply by aaronpowell** [reply]
Pretty sure Strapi does too

**Reply by dcousens** [reply]
Thanks for the followup :heart_decoration:
